pynestml/codegeneration/code_generator.py

```diff
@@ -58,10 +58,12 @@ def generate_neuron_code(self, neuron: ASTNeuron) -> None:
 
     def set_options(self, options: Mapping[str, Any]):
         if not "_default_options" in dir(self.__class__):
-            assert "Code generator class \"" + str(self.__class__) + "\" does not support setting options."
+            print("Warning: Code generator class \"" + str(self.__class__) + "\" does not support setting options.")
         for k in options.keys():
-            assert k in self.__class__._default_options, "Option \"" + str(k) + "\" does not exist in code generator"
-        self._options.update(options)
+            if k in self.__class__._default_options:
+                self._options[k] = options[k]
+            else:
+                print("Warning: Option \"" + str(k) + "\" does not exist in code generator")
 
     def get_option(self, k):
         return self._options[k]
```

Comment:
I think the following would be better: `self._options.get(k, None)`, in case if the dictionary doesn't contain the key.

Reply:
I'd rather it throw an exception. What if there exists a known key but with value None?